### PR TITLE
MB-32183 - new Levenshtein automaton

### PR DESCRIPTION
**Pull request by sreekanth-cb**
Adopting new levenshtein automaton

### CODE REVIEW

**Comment by abhinavdangeti** [comment]
Hey @sreekanth-cb , any particular reason you've force pushed a branch "newlev_adoption" to blevesearch/bleve?

**Comment by sreekanth-cb** [comment]
I was just guessing that a commit -amend would help re triggering the travis ci with the latest bleve changes for ci fixes.. But may be just re running/restarting those jobs we enough..

File: index/scorch/snapshot_index.go [hunk 1]
```diff
@@ -46,6 +51,8 @@ var reflectStaticSizeIndexSnapshot int
 func init() {
 	var is interface{} = IndexSnapshot{}
 	reflectStaticSizeIndexSnapshot = int(reflect.TypeOf(is).Size())
+	lb1, lbErr1 = lev2.NewLevenshteinAutomatonBuilder(1, true)
```
Comment (steveyen) [inline index/scorch/snapshot_index.go]: another way of doing this is panic()'ing if the lev2 returns error here in init()?  that way, later code doesn't need to check for lb1 != nil

File: index/scorch/snapshot_index.go [hunk 2]
```diff
@@ -194,9 +201,26 @@ func (i *IndexSnapshot) FieldDictRegexp(field string,
...
+		}
+		return nil, lbErr1
+
+	} else if fuzziness == 2 {
+		if lb2 != nil {
+			return lb2.BuildDfa(term, fuzziness)
+		}
+		return nil, lbErr2
+	}
+	return nil, nil
```
Comment (steveyen) [inline index/scorch/snapshot_index.go]: feels like return an error here would be better if fuzziness is unsupported

**Comment by abhinavdangeti** [comment]
> I was just guessing that a commit -amend would help re triggering the travis ci with the latest bleve changes for ci fixes.. But may be just re running/restarting those jobs was enough..

Oh I'm not talking about travis at all. Our usual process is - we create a branch in the fork of bleve, and create a PR with it. Here you have force pushed your private branch to the main bleve repo, and are attempting to pull that down to the master branch. We'll need to make sure that once this change is merged, the branch _newlev_adoption_ is deleted in the main repo.

**Comment by mschoch** [comment]
I get nervous hearing "guess" and "force push" in the same conversation :)

**Comment by sreekanth-cb** [comment]
Ah.. I used to do PRs in both ways, like a direct branch merge and from a forked repo. I do the direct branch merge when my forked repo is having too many other changes and when I push a branch - it shows with so many other commit history. It happens even when I 
-forked bleve repo, but made few changes for some other fixes.
-checkout master
-pull the latest
-create a new branch and make fix changes and push to bleve master..

Being lazy to fix this issue, I create a branch in bleve and directly open a PR against master.. Once in a while I do clean those stale branches.Hence I am not seeing many (only 1/2) from me as of now.

**Comment by sreekanth-cb** [comment]
closing this PR, as the PR from a branch against bleve master won't help in taking other fixes in master while running the CI. Will create another afresh..

**Pull request closed without merging**